service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,117 @@
+public class UploadDocumentsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public Map<String, Object> updateCaseDocuments(CaseData caseData, CaseData caseDataBefore) {
+        List<Element<DocumentSocialWorkOther>> otherSocialWorkDocuments = setUpdatedByAndDateAndTimeOnDocuments(
+                caseData.getOtherSocialWorkDocuments(), caseDataBefore.getOtherSocialWorkDocuments());
+
+        Document socialWorkChronologyDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkChronologyDocument(), caseDataBefore.getSocialWorkChronologyDocument());
+
+        Document socialWorkStatementDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkStatementDocument(), caseDataBefore.getSocialWorkStatementDocument());
+
+        Document socialWorkAssessmentDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkAssessmentDocument(), caseDataBefore.getSocialWorkAssessmentDocument());
+
+        Document socialWorkCarePlanDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkCarePlanDocument(), caseDataBefore.getSocialWorkCarePlanDocument());
+
+        Document socialWorkEvidenceTemplateDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkEvidenceTemplateDocument(), caseDataBefore.getSocialWorkEvidenceTemplateDocument());
+
+        Document thresholdDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getThresholdDocument(), caseDataBefore.getThresholdDocument());
+
+        Document checklistDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getChecklistDocument(), caseDataBefore.getChecklistDocument());
+
+        CourtBundle courtBundleDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle());
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents_socialWorkOther", otherSocialWorkDocuments);
+        updatedCaseData.put("documents_socialWorkChronology_document", socialWorkChronologyDocument);
+        updatedCaseData.put("documents_socialWorkStatement_document", socialWorkStatementDocument);
+        updatedCaseData.put("documents_socialWorkAssessment_document", socialWorkAssessmentDocument);
+        updatedCaseData.put("documents_socialWorkCarePlan_document", socialWorkCarePlanDocument);
+        updatedCaseData.put("documents_socialWorkEvidenceTemplate_document", socialWorkEvidenceTemplateDocument);
+        updatedCaseData.put("documents_threshold_document", thresholdDocument);
+        updatedCaseData.put("documents_checklist_document", checklistDocument);
+        updatedCaseData.put("courtBundle", courtBundleDocument);
+
+        return updatedCaseData;
+    }
+
+    public <T extends DocumentMetaData> List<Element<T>> setUpdatedByAndDateAndTimeOnDocuments(
```

Comment:
Do you think that this method is essentially being replicated in each of the services you have altered? If so do you think we could extract it to a helper class which and the method could have the additional parameters of `LocalDateTime timeOfUpload` and `String uploadedBy`? Not now but maybe a future bit of work that you could do in the background whilst working on other things.

Reply:
100% agree. That is the reason I made it generic hoping that it can be used in other places. I didn't want to do all that in this PR. I will create a tech debt for it now.

Reply:
https://tools.hmcts.net/jira/browse/FPLA-2233